src/amplifyp/gui/app.py

```
@@ -49,40 +49,51 @@ def main(page: ft.Page) -> None:
         def confirm_dismiss(e: ft.ControlEvent) -> None:
-            confirm_dialog.open = False
-            page.update()
+            dialog = getattr(page, "_confirm_dialog", None)
+            if dialog:
+                dialog.open = False
+                page.update()
+
+        async def confirm_exit_async() -> None:
+            try:
+                await page.window.destroy()
+            except Exception:  # noqa: S110
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)